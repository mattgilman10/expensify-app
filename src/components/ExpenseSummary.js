import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

// my version
// export const ExpenseSummary = ({expenseCount, expensesTotal}) => (
//     <div>
//         {
//             props.expenses.length === 1 ? (
//                 <p>Viewing 1 expense totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')} </p>
//             ) : (
//                 <p>Viewing {props.expenses.length} expenses totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')} </p>
//             )
//         }
//     </div>
// );


export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
    );
};



const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
