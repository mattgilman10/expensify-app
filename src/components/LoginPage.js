import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button onClick={startLogin} className='button'>Login with Google</button>
        </div>
        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);













// const LoginPage = () => (
//     <div>
//         <h1>Login Page</h1>
//         <form>
//             <input 
//                 type="text"
//                 placeholder="Name"
//             />
//             <input 
//                 type="text"
//                 placeholder="Password"
//             />
//             <Link to="/dashboard">
//                 <button>
//                     Login
//                 </button>
//             </Link>
//         </form>
//     </div>
// );