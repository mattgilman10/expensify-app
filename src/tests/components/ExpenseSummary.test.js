import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test("should render ExpenseSummary with multiple expenses", () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test("should render expense list with singular expense", () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={23559290}/>);
    expect(wrapper).toMatchSnapshot();
});