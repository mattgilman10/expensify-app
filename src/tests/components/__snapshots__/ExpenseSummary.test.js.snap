// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $2.35
</h1>
`;

exports[`should render expense list with singular expense 1`] = `
<h1>
  Viewing 
  23
   
  expenses
   totalling 
  $235,592.90
</h1>
`;
